chcemvediet/apps/inforequests/admin.py

```diff
@@ -3,37 +3,38 @@
 import datetime
 
 from django.contrib import admin
+from django.contrib.admin.actions import delete_selected
 from django.contrib.admin.utils import NestedObjects
 from django.core.exceptions import PermissionDenied
-from django.db import router
+from django.db import router, transaction
 from django.forms.models import BaseInlineFormSet
 from django.utils.html import format_html
 
 from poleno.utils.date import local_today
-from poleno.utils.misc import decorate
+from poleno.utils.misc import decorate, squeeze
 from poleno.utils.admin import (simple_list_filter_factory, admin_obj_format,
-                                ReadOnlyAdminInlineMixin, NoBulkDeleteAdminMixin)
+                                ReadOnlyAdminInlineMixin)
 from chcemvediet.apps.inforequests.constants import ADMIN_EXTEND_SNOOZE_BY_DAYS
 
 from .models import Inforequest, InforequestDraft, InforequestEmail, Branch, Action
 
 
 class DeleteNestedInforequestEmailAdminMixin(admin.ModelAdmin):
 
-    def get_inforequest(self, obj):
-        raise NotImplementedError
-
-    def nested_inforequestemail_queryset(self, obj):
+    def nested_inforequestemail_queryset(self, objs):
         using = router.db_for_write(self.model)
         collector = NestedObjects(using)
-        collector.collect([obj])
+        collector.collect(objs)
         to_delete = collector.nested()
-        inforequest = self.get_inforequest(obj)
         actions = [obj for obj in self.nested_objects_traverse(to_delete)
                    if isinstance(obj, Action)]
-        emails = [action.email for action in actions if action.email]
-        inforequestemails_qs = InforequestEmail.objects.filter(inforequest=inforequest,
-                                                               email__in=emails)
+        inforequestemails_qs = InforequestEmail.objects.none()
+        for action in actions:
+            if action.email:
+                inforequestemails_qs |= InforequestEmail.objects.filter(
+                        inforequest=action.branch.inforequest,
+                        email=action.email,
+                )
```

Comment:
Tu som rozmyslal, ci by nebolo prehladnejsie pouzit `Q(inforequest=action.branch.inforequest, email=action.email)`. A potom pouzit [`reduce`] s [`or_`] operatorom na zlepenie pola Q vyrazov do jednej podmienky.

[`reduce`]: https://docs.python.org/2/library/functions.html#reduce
[`or_`]: https://docs.python.org/2/library/operator.html#operator.or_

Cize by sme mali nieco cca ako:
```
inforequestemails_conditions = [Q(...) for action in actions]
inforequestemails_qs = InforequestEmail.objects.filter(reduce(or_, inforequestemails_conditions))
```
Ale nie som si isty, ktory pristup je prehladnejsi. Co si myslis? Ak sa nemyslim, oba pristupy vygeneruju rovnaku query.

Reply:
Ja osobne preferujem operator `|=` s `for cyklom, ale oba sposoby su prehladne.